KomaMRIBase/src/timing/TimeStepCalculation.jl

```diff
@@ -0,0 +1,163 @@
+const MIN_RISE_TIME = 1e-10
+"""
+    array_of_ranges = kfoldperm(N, k; type="random", breaks=[])
+
+Divides a list of indices 1:`N` (which is in your imagination) into `k` groups.
+
+!!! note
+    It is possible to predifine some break points at specific indices with the `breaks`
+    keyword, in this case the number of groups could increase. This is useful to define
+    start and end indices of RF pulses to separate the simulation into excitation and
+    precession computations.
+
+# Arguments
+- `N`: (`::Int64`) the number of elements to be ordered (of an imaginary array 1:`N`)
+- `k`: (`::Int64`) the number of groups to divide the `N` elements
+
+# Keywords
+- `type`: (`::String`, `="random"`, opts: [`"random"`, `"ordered"`]) the order type option.
+    If random, then the indices of the groups are unordered. If "ordered", then the indices
+    of the groups are sorted in an incremental order
+- `breaks`: (`::Vector{Int64}`, `=[]`) the array of indices where predefined break points
+    are placed
+
+# Returns
+- `array_of_ranges`: (`::Vector{UnitRange{Int64}}`) the array that contains ranges of
+    different groups (the aim target are `k` groups, but this could be increased by adding
+    elements in the `breaks` input array)
+
+# Examples
+``` julia-repl
+julia> kfoldperm(20, 3; type="ordered")
+3-element Vector{UnitRange{Int64}}:
+ 1:7
+ 8:14
+ 15:20
+
+julia> kfoldperm(20, 3; type="ordered", breaks=[3])
+4-element Vector{UnitRange{Int64}}:
+ 1:2
+ 3:7
+ 8:14
+ 15:20
+```
+"""
```

Comment:
This docstring is maybe too long for a function that is not even exported. Also I believe this can be replaced by a function in Base called partition or something like that.

Reply:
`using Base.Iterators: partition`, or this is the package, read this: https://julialang.org/blog/2023/07/PSA-dont-use-threadid/